lms/views/admin/application_instance.py

```diff
@@ -294,9 +294,35 @@ def show_instance(self):
     def update_instance(self):
         ai = self._get_ai_or_404(**self.request.matchdict)
 
-        for ai_field in ["lms_url", "deployment_id"]:
-            if value := self.request.params.get(ai_field, "").strip():
-                setattr(ai, ai_field, value)
+        if lms_url := self.request.params.get("lms_url", "").strip():
+            ai.lms_url = lms_url
+
+        if deployment_id := self.request.params.get("deployment_id", "").strip():
+            if ai.lti_version == "1.3.0":
+                try:
+                    duplicated_ai = (
```

Comment:
Is this the `existing_ai` rather than the `duplicated_ai`. It sounds like our `ai` is the one which would be duplicating?

This is also the language used in the tests below.